xrpl/model/transactions/payment.go

```diff
@@ -57,8 +57,46 @@ type Payment struct {
 	SendMax types.CurrencyAmount `json:",omitempty"`
 }
 
-func (*Payment) TxType() TxType {
-	return PaymentTx
+func (p *Payment) Flatten() map[string]interface{} {
```

Comment:
So am I correct to say that `Flatten` converts a `struct` to a `map[string]interface{}` ?